decoder/src/log/format/mod.rs

```diff
@@ -265,42 +272,144 @@ enum Record<'a> {
 }
 
 #[derive(Debug)]
+#[non_exhaustive]
 pub enum FormatterFormat<'a> {
-    Default { with_location: bool },
+    /// The classic defmt two-line format.
+    ///
+    /// Looks like:
+    ///
+    /// ```text
+    /// INFO This is a log message
+    /// └─ test_lib::hello @ /Users/jonathan/Documents/knurling/test-lib/src/lib.rs:8
+    /// ```
+    Default {
+        with_location: bool,
+    },
+    /// A one-line format.
+    ///
+    /// Looks like:
+    ///
+    /// ```text
+    /// [INFO ] This is a log message (crate_name test-lib/src/lib.rs:8)
+    /// ```
+    OneLine {
+        with_location: bool,
+    },
     Custom(&'a str),
 }
 
+impl FormatterFormat<'static> {
+    /// Parse a string into a choice of [`FormatterFormat`].
+    ///
+    /// Unknown strings return `None`.
+    pub fn from_string(s: &str, with_location: bool) -> Option<FormatterFormat<'static>> {
+        match s {
+            "default" => Some(FormatterFormat::Default { with_location }),
+            "oneline" => Some(FormatterFormat::OneLine { with_location }),
+            _ => None,
+        }
+    }
+
+    /// Get a list of valid string names for the various format options.
+    ///
+    /// This will *not* include an entry for [`FormatterFormat::Custom`] because
+    /// that requires a format string argument.
+    pub fn get_options() -> impl Iterator<Item = &'static str> {
+        ["default", "oneline"].iter().cloned()
+    }
+}
+
 impl Default for FormatterFormat<'_> {
     fn default() -> Self {
-        FormatterFormat::Default {
+        FormatterFormat::OneLine {
```

Comment:
So we are changing the default now?

Reply:
Oops, no, that's a mistake - I was just testing.